--- /dev/null
+++ ./PKGBUILD
@@ -53,7 +53,8 @@
         'brave-browser.desktop'
         "chromium-launcher-$_launcher_ver.tar.gz::https://github.com/foutrelis/chromium-launcher/archive/v$_launcher_ver.tar.gz"
         "https://github.com/stha09/chromium-patches/releases/download/${patchset_name}/${patchset_name}.tar.xz"
-        "chromium-no-history.patch")
+        "chromium-no-history.patch"
+        "gtk-dark-mode-switch-fix.patch")
 arch_revision=7cd2f08852ca3b88237a5c563eb8cfc5e471e489
 Patches="
         fix-crash-in-ThemeService.patch
@@ -77,6 +78,7 @@
             '86859c11cfc8ba106a3826479c0bc759324a62150b271dd35d1a0f96e890f52f'
             '171525009003a9ed1182cfcb6f407d7169d9a731a474304e263029376719f55a'
             'ea3446500d22904493f41be69e54557e984a809213df56f3cdf63178d2afb49e'
+            'd8da17f60c8a66ee57e7d080f7113a05e049288abbe37b1f01bb3e2220e2657f'
             '3cfe46e181cb9d337c454b5b5adbf5297052f29cd617cdee4380eeb1943825d8'
             '59a59a60a08b335fe8647fdf0f9d2288d236ebf2cc9626396d0c4d032fd2b25d'
             '50133dd196d288ad538bb536aa51dccd6cb4aacfd9a60160f77e8fb16034b460'
@@ -158,6 +160,9 @@
   msg2 "Apply Chromium patches..."
   cd src/

+  # Apply gtk dark mode switch patch
+  patch -Np1 -i ../../gtk-dark-mode-switch-fix.patch
+
   # https://crbug.com/893950
   sed -i -e 's/\<xmlMalloc\>/malloc/' -e 's/\<xmlFree\>/free/' \
     third_party/blink/renderer/core/xml/*.cc \
